### PR TITLE
Fix SMSG_SPELLLOGEXECUTE structure and data

### PR DESCRIPTION
**Pull request by Laizerox**
Cleanup spell log execution
Simplify a lot of code
Fix almost all issues with combat log?

### CODE REVIEW

**Comment by Warlockbugs** [comment]
I can confirm that extra attacks and counterspell are displayed properly with this patch.

File: src/game/Spell.cpp [hunk 1]
```diff
@@ -6910,3 +6839,53 @@ void Spell::GetSpellRangeAndRadius(SpellEffectIndex effIndex, float& radius, uin
...
+    logEffect << param4;
+
+    m_executeLog[effIndex].push_back(logEffect);
+}
+
+template void Spell::LogEffectExecute(uint8, uint32);
+template void Spell::LogEffectExecute(uint8, PackedGuid);
+template void Spell::LogEffectExecute(uint8, PackedGuid, uint32);
+template void Spell::LogEffectExecute(uint8, PackedGuid, int32, int32);
+template void Spell::LogEffectExecute(uint8, PackedGuid, uint32, uint32, float);
```
Comment (RAlexis) [inline src/game/Spell.cpp]: New line

**Comment by Phatcat** [comment]
@Laizerox merge errors you may wanna look at after recent commits

**Comment by Laizerox** [comment]
@Phatcat This is still kind of WIP and I will merge it in the next update likely.

File: src/game/Spell.cpp (hunk 1 above)
Comment (mityada) [inline src/game/Spell.cpp]: Why not use variadic templates here?

```
void FillExecuteLog(ByteBuffer &buffer)
{
}

template <typename P, typename ... Params>
void FillExecuteLog(ByteBuffer &buffer, P param, Params ... params)
{
    buffer << param;
    FillExecuteLog(buffer, params...);
}

template <typename ... Params>
void Spell::LogEffectExecute(uint8 effIndex, Params ... params)
{
    ByteBuffer logEffect;
    FillExecuteLog(logEffect, params...);

    m_executeLog[effIndex].push_back(logEffect);
}
```

**Comment by Laizerox** [comment]
@mityada it was the plan but after thinking of it only overload that would require it is the one with 1 param.

Also we would still need to declare possible templates and templates can be too ambigous. Which means my final version which I havent push yet will have 5 overloads instead.

**Pull request closed without merging**